echo_server.py

```diff
@@ -1,40 +1,95 @@
-from __future__ import print_function
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import email.utils
 import socket
-import os
 
 
-s = socket.socketpair(socket.AF_UNIX, socket.SOCK_STREAM)
-print ('Socket created')
+def response_ok():
+    first_line = 'HTTP/1.1 200 OK'
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    content_header = 'Content-Type: text/html'
+    body = '''<!DOCTYPE html>\n<html lang="en">\n<head>\n<meta charset="utf-8">\n</head>\n<body>\n{}</body>\n</html>'''.format('200 OK')
+    response_list = [first_line, timestamp, content_header, '', body, '\r\n']
+    return '\r\n'.join(response_list).encode('utf-8')
 
 
-def socketpair():
-    done = False
-    buffer_size = 16
-    response = ''
+def response_error(error):
+    first_line = 'HTTP/1.1 {} {}'.format(error.code, error.msg)
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    content_header = 'Content-Type: text/plain'
+    body = '{} {}'.format(error.code, error.msg)
+    body = '''<!DOCTYPE html>\n<html lang="en">\n<head>\n<meta charset="utf-8">\n</head>\n<body>\n{}</body>\n</html>'''.format(body)
+    response_list = [first_line, timestamp, content_header, '', body, '\r\n']
+    return '\r\n'.join(response_list)
+
+
+def parse_request(request):
+    first_line = request.splitlines()[0]
+    first_line = first_line.split(" ")
```

Comment:
If the amount of space between the elements of the first line varies from one space exactly, this will fail in some pretty interesting ways.  If you mean to say "split on whitespace" then just omit the argument to split:

```python
>>> bar = "one  two  three"
>>> bar.split(" ")
['one', '', 'two', '', 'three']
>>> bar.split()
['one', 'two', 'three']
>>>
```